addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,476 @@
+                        String response = updateKonnectedModule();
+                        logger.trace("The response from the konnected module with thingID {} was {}",
+                                getThing().getUID().toString(), response);
+                        if (response == null) {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "Unable to communicate with Konnected Module confirm settings.");
+                        } else {
+                            updateStatus(ThingStatus.ONLINE);
+                        }
+                    } catch (IOException e) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                    }
+                    value = false;
+                }
+            }
+        }
+
+        super.handleConfigurationUpdate(configurationParameters);
+        try {
+            String response = updateKonnectedModule();
+            logger.trace("The response from the konnected module with thingID {} was {}",
+                    getThing().getUID().toString(), response);
+            if (response == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Unable to communicate with Konnected Module confirm settings.");
+            } else {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            logger.trace("There was an IOEcption Error thrown during HandleConfigurationUpdate()");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(() -> {
+            try {
+                checkConfigruation();
+            } catch (ConfigValidationException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            }
+            this.moduleIpAddress = this.getThing().getProperties().get(HOST).toString();
+            // add the isActuator elements to the boolean array
+            logger.debug("Setting up Konnected Module WebHook");
+            try {
+                webHookServlet.activate(this);
+            } catch (KonnectedWebHookFail e) {
+                logger.trace("there was an error activating the servelet: {}", e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            }
+            try {
+                String response = updateKonnectedModule();
+                if (response == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Unable to communicate with Konnected Module confirm settings or readd thing.");
```

Comment:
Is this a proper sentence? Try reformulating it.
It also seems that you are handling all kinds of different errors here at once - this isn't necessarily very helpful for the user wrt analysing the problem.

Reply:
Ok, if you are handling all "expected" problems like connection timeouts etc. elsewhere, this "something went wrong" catch is just fine. Maybe nonetheless reforumulate it to simply "Unable to communicate with Konnected Module".

Reply:
reformulated